java/adapter/avro/src/main/java/org/apache/arrow/consumers/AvroFixedConsumer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.consumers;
+
+import java.io.IOException;
+
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.FixedSizeBinaryVector;
+import org.apache.arrow.vector.complex.impl.FixedSizeBinaryWriterImpl;
+import org.apache.arrow.vector.complex.writer.FixedSizeBinaryWriter;
+import org.apache.avro.io.Decoder;
+
+import io.netty.buffer.ArrowBuf;
+
+/**
+ * Consumer which consume fixed type values from avro decoder.
+ * Write the data to {@link org.apache.arrow.vector.FixedSizeBinaryVector}.
+ */
+public class AvroFixedConsumer implements Consumer {
+
+  private final FixedSizeBinaryWriter writer;
+  private final FixedSizeBinaryVector vector;
+
+  private final ArrowBuf cacheBuffer;
+  private final byte[] reuseBytes;
+
+  /**
+   * Instantiate a AvroFixedConsumer.
+   */
+  public AvroFixedConsumer(FixedSizeBinaryVector vector, int size) {
+    this.vector = vector;
+    this.writer = new FixedSizeBinaryWriterImpl(vector);
+    cacheBuffer = vector.getAllocator().buffer(size);
+    reuseBytes = new byte[size];
+  }
+
+  @Override
+  public void consume(Decoder decoder) throws IOException {
+    decoder.readFixed(reuseBytes);
+    cacheBuffer.setBytes(0, reuseBytes);
```

Comment:
I think I keep asking this, but is it possible to avoid the cachebuffer and instead copy the bytes directly into  the FixedSizedBinaryVector?

Reply:
Right, fixed.